adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotations.scala

```diff
@@ -0,0 +1,140 @@
+    WARNING_SEQUENCE_NOT_AVAILABLE.name() -> WARNING_SEQUENCE_NOT_AVAILABLE,
+    WARNING_TRANSCRIPT_INCOMPLETE.name() -> WARNING_TRANSCRIPT_INCOMPLETE,
+    WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS.name() -> WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS,
+    WARNING_TRANSCRIPT_NO_START_CODON.name() -> WARNING_TRANSCRIPT_NO_START_CODON,
+    INFO_REALIGN_3_PRIME.name() -> INFO_REALIGN_3_PRIME,
+    INFO_COMPOUND_ANNOTATION.name() -> INFO_COMPOUND_ANNOTATION,
+    INFO_NON_REFERENCE_ANNOTATION.name() -> INFO_NON_REFERENCE_ANNOTATION,
+    // message code -> enum
+    "E1" -> ERROR_CHROMOSOME_NOT_FOUND,
+    "E2" -> ERROR_OUT_OF_CHROMOSOME_RANGE,
+    "W1" -> WARNING_REF_DOES_NOT_MATCH_GENOME,
+    "W2" -> WARNING_SEQUENCE_NOT_AVAILABLE,
+    "W3" -> WARNING_TRANSCRIPT_INCOMPLETE,
+    "W4" -> WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS,
+    "W5" -> WARNING_TRANSCRIPT_NO_START_CODON,
+    "I1" -> INFO_REALIGN_3_PRIME,
+    "I2" -> INFO_COMPOUND_ANNOTATION,
+    "I3" -> INFO_NON_REFERENCE_ANNOTATION
+  )
+
+  private def parseEffects(s: String): List[String] = {
+    s.split("&").toList
+  }
+
+  private def parseMessages(s: String): List[VariantAnnotationMessage] = {
+    // todo: haven't seen a delimiter here, assuming it is also '&'
+    s.split("&").map(MESSAGES.get(_)).toList.flatten
+  }
+
+  private def parseFraction(s: String): (Option[Integer], Option[Integer]) = {
+    if ("".equals(s)) {
+      return (None, None)
+    }
+    val tokens = s.split("/")
+    tokens.length match {
+      case 0 => (None, None)
+      case 1 => (Some(Integer.parseInt(tokens(0))), None)
+      case _ => (Some(Integer.parseInt(tokens(0))), Some(Integer.parseInt(tokens(1))))
+    }
+  }
+
+  private def notEmpty(s: String): Boolean = Option(s) match {
+    case None    => false
+    case Some(s) => s.length > 0
+  }
+
+  private[converters] def parseTranscriptEffect(s: String): Seq[TranscriptEffect] = {
+    val fields = s.split("\\|", -1)
+    if (fields.length < 16) {
+      if (log.isWarnEnabled) log.warn("Invalid VCF ANN attribute, expected at least 16 fields, found {}: {}", fields.length, fields)
+      return Seq()
+    }
+
+    // simple and ignored fields
+    val (alternateAllele, annotationImpact, geneName, geneId, featureType,
+      featureId, biotype, transcriptHgvs, proteinHgvs, distance) =
+      (fields(0), fields(2), fields(3), fields(4), fields(5),
+        fields(6), fields(7), fields(9), fields(10), fields(14))
+
+    // multivalued fields
+    val effects = parseEffects(fields(1))
+    val (rank, total) = parseFraction(fields(8))
+    val (cdnaPosition, cdnaLength) = parseFraction(fields(11))
+    val (cdsPosition, cdsLength) = parseFraction(fields(12))
+    val (proteinPosition, proteinLength) = parseFraction(fields(13))
+    val messages = parseMessages(fields(15))
+
+    val te = TranscriptEffect.newBuilder()
+    if (notEmpty(alternateAllele)) te.setAlternateAllele(alternateAllele)
```

Comment:
If you do https://github.com/bigdatagenomics/adam/pull/1069/files#r71388866, then you can eliminate all of these if statements.